packages/image_picker/image_picker/ios/Classes/FLTImagePickerPlugin.m

```diff
@@ -197,6 +245,14 @@ - (void)checkPhotoAuthorization {
           dispatch_async(dispatch_get_main_queue(), ^{
             [self showPhotoLibrary];
           });
+        } else if (@available(iOS 14, *)) {
+          if (status == PHAuthorizationStatusLimited) {
+            dispatch_async(dispatch_get_main_queue(), ^{
+              [self showLimitedPhotoLibrary];
+            });
+          } else {
+            [self errorNoPhotoAccess:status];
```

Comment:
I think it was a mistake in the intial design of this code, but everything in the block should be dispatched to main, so 

```objc
dispatch_async(dispatch_get_main_queue(), ^{
   if (status == PHAuthorizationStatusAuthorized) {
     [self showPhotoLibrary];
    } else if (@available(iOS 14, *)) {
     if (status == PHAuthorizationStatusLimited) {
         [self showLimitedPhotoLibrary];
     } else {
         [self errorNoPhotoAccess:status];
     }
 }  else {
    [self errorNoPhotoAccess:status];
}});
```

While you are here, could you make this fix ;)

Reply:
Done!

Reply:
hmm you commented "done" but I didn't see the change. Was the change not pushed?

Reply:
I guess. It is done now!